Add babel-preset-env standalone.

yavorsky opened the pull request:
Have tested with the https://babeljs.io/7 REPL.

@Daniel15 We need to consider npm packages, so I'll give npm access to babel-preset-env-standalone and we can unpublish old and publish the new one.

existentialism commented on packages/babel-preset-env-standalone/README.md:
```diff
@@ -0,0 +1,5 @@
+babel-preset-env-standalone
+================
+
+babel-preset-env-standalone is a standalone build of [babel-preset-env](https://github.com/babel/babel-preset-env) for use in non-Node.js environments, including browsers. Epsecially with [babel-standalone](https://github.com/babel/babel-standalone). 
```
Nit: Epsecially -> Especially

Maybe:

It is typically used alongside [babel-standalone]().

yavorsky replied:
@existentialism 👍

yavorsky replied:
Maybe rewrite the whole README to match [babili's format](https://github.com/babel/babel-standalone/blob/master/packages/babili-standalone/README.md) ?

Daniel15 replied:
Yeah let's keep the style the same as the Babili one, except you can probably omit the "But why?!" section. You can also omit the CDNJS link until it's actually available on CDNJS.

Daniel15 commented:
Thanks for submitting a pull request! I'll take a look soon.

Sent from my phone.

On Jul 25, 2017 6:32 AM, "Artem Yavorsky" <notifications@github.com> wrote:

> *@yavorsky* commented on this pull request.
> ------------------------------
>
> In packages/babel-preset-env-standalone/README.md
> <https://github.com/babel/babel-standalone/pull/90#discussion_r129304734>:
>
> > @@ -0,0 +1,5 @@
> +babel-preset-env-standalone
> +================
> +
> +babel-preset-env-standalone is a standalone build of [babel-preset-env](https://github.com/babel/babel-preset-env) for use in non-Node.js environments, including browsers. Epsecially with [babel-standalone](https://github.com/babel/babel-standalone).
>
> Maybe rewrite the whole README to match babili's format
> <https://github.com/babel/babel-standalone/blob/master/packages/babili-standalone/README.md>
> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/babel/babel-standalone/pull/90#discussion_r129304734>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFnHQj8SZF0543B9NCpE4XnQUx9H5W8ks5sRe5ogaJpZM4OiYxP>
> .
>

Daniel15 commented on package.json:
```diff
@@ -142,7 +143,7 @@
     "transpiler"
   ],
-  "author": "Daniel Lo Nigro \u003cdaniel@dan.cx\u003e (http://dan.cx/)",
+  "author": "Daniel Lo Nigro <daniel@dan.cx> (http://dan.cx/)",
```
Hmm. This is okay, but the automated upgrade script will just change the `<` back to `\u0003` ¯\\\_(ツ)\_\/¯

Daniel15 commented on packages/babel-preset-env-standalone/package.json:
```diff
@@ -0,0 +1,29 @@
+  "bugs": {
+    "url": "https://github.com/babel/babel-standalone/issues"
+  },
+  "homepage": "https://babeljs.io/repl"
```
Use `https://babel-standalone.dan.cx/` as the URL please (although, it just redirects to the Github project)

Daniel15 commented on src/build-babel-preset-env.js:
```diff
@@ -0,0 +1,218 @@
+import pluginList from "babel-preset-env/data/plugins.json";
```
Is there an advantage to having this in a separate file? What about just putting this directly into `babel-preset-env`?

yavorsky replied:
Just to separate babel-preset-env stuff with the scripts for building standalone version. But after last changes we have just 3 lines for the last, so I'll revert this.

Daniel15 commented on src/babel-preset-env.js:
```diff
@@ -0,0 +1,18 @@
+
+registerPreset('env', require('./build-babel-preset-env.js'));
+
+export function transform (code, options = {}) {
```
Do you think this function is beneficial? People can just call `Babel.transform` directly and just pass `env` as the preset. I don't think we need another separate entry point just for `babel-preset-env`. Babili is special since it's available as a separate CLI app.

Daniel15 commented on packages/babel-preset-env-standalone/package.json:
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "babel-preset-env-standalone",
```
This name is a bit long. What do you think of calling it `babel-standalone-env` instead?

Daniel15 commented:
We'll also need to update the auto-update script (`scripts/update.ps1`) to handle updating the env package, but we can do that separately and just manually publish updates for now 😃

yavorsky commented:
@Daniel15 
I think all requested changes are ready:

1. Renamed to babel-env-standalone
2. Using single build file.
3. Removed transform for this file.
4. Updated README, both `package.json`s.
5. Added test cases for expicit targets, browsers and usBuiltIns: usage.
6. Updated example: browsers field with throttling, useBuilsIns(usage/entry options), error field to show when incorrect browser query was passed and debug for each transformation.

yavorsky commented:
https://github.com/babel/babel-preset-env/pull/379 was merged, so we can greatly reduce `src/babel-preset-env.js`. Going to update in few hours.

Daniel15 commented:
Sorry for the delay, I'll take a look at this again shortly.

yavorsky commented:
@Daniel15 going to resolve merge conflicts today.
I've reduced src/babel-preset-env after https://github.com/babel/babel-preset-env/pull/379 was merged. Also I've mocked availablePlugins with our plugins registered in the babel-standalone, so bundle became much smaller.

Daniel15 commented:
Thanks.

Note that recently I merged `babel-standalone` into the Babel repo itself: https://github.com/babel/babel/pull/6029. I'm still not sure what I'll do with env and `babili-standalone` so they're still fine to live in this repo for now.

Daniel15 commented:
It looks like this broke the build somehow... Could you please take a look? https://travis-ci.org/babel/babel-standalone/builds/264183835?utm_source=email&utm_medium=notification

yavorsky commented:
@Daniel15 Hm, strange, initially, it was built successfully. Going to resolve this now.
